pkg/controller/servicemesh/webhookca/controller.go

```diff
@@ -45,20 +46,28 @@ const (
 // secretNames determines priority.
 var autoRegistrationMap = map[string]CABundleSource{
 	galleyWebhookNamePrefix: &SecretCABundleSource{
-		SecretNames: []string{galleySecretName},
-		Key:         common.IstioRootCertKey,
+		SecretNameAndKeyPairs: map[string]string{
+			galleySecretName: common.IstioRootCertKey,
+		},
 	},
 	sidecarInjectorWebhookNamePrefix: &SecretCABundleSource{
-		SecretNames: []string{sidecarInjectorSecretName},
-		Key:         common.IstioRootCertKey,
+		SecretNameAndKeyPairs: map[string]string{
+			sidecarInjectorSecretName: common.IstioRootCertKey,
+		},
 	},
 	istiodWebhookNamePrefix: &SecretCABundleSource{
-		SecretNames: []string{istiodCustomCertSecretName, istiodSecretName},
-		Key:         common.IstiodCertKey,
+		SecretNameAndKeyPairs: map[string]string{
+			istiodCustomCertSecretName:  common.IstiodCertKey,
+			istiodSecretName:            common.IstiodCertKey,
+			istiodCertManagerSecretName: common.IstiodCertManagerCertKey,
```

Comment:
This line, and the corresponding line below, are crucial changes in this PR. The operator was not able to reconcile webhooks, because it couldn't find proper secret.

Reply:
Yes these are crucial and it would work with this code change, but we are working on a fix that doesn't require you to change this. we will show you it in a few days tops once it's out. if there are any downsides to the alternative solution we can come back to this one. There is risk with changing this key for other providers i think.

Reply:
Actually after consulting with other istio-csr maintainers, we think that perhaps this change is the better approach, in the meanwhile it wouldn't be unreasonable to add some docs with the steps for version v2.3.0 to just run a pre-installation step to explicitely create the secret in the istio-system namespace manually, it is best practice to get this secret and statically set it in cert-manager namespace anyways, so manually doing it would also be fine, we can even keep doing that for version v2.3.1 if we wanted but i am happy to accept these changes as well.

The only risk we need to make sure is that by changing this we break people who are manually loading in certs etc..

Reply:
> in the meanwhile it wouldn't be unreasonable to add some docs with the steps for version v2.3.0 to just run a pre-installation step to explicitely create the secret in the istio-system namespace manually

Good idea. I completely forgot, that this secret could be customized by users manually to make it work with 2.3.0.

Reply:
Taking into account that users can customize secret to make it work with istio-operator 2.3.0, I have to revert removal of `pilotSecretName` and `rootCAConfigMapName`. However, I'm still thinking why we have these values configurable.
1. `rootCAConfigMapName` is no longer configurable in istio-csr since the release v.0.3.0, so shouldn't we deprecate it?
2. If it comes to `pilotSecretName`, I can imagine that it will become configurable in the future, but what if secret keys will change and we don't enable users to configure them?

Instead of making these values configurable in istio-operator, they should be configurable in istio-csr, so that users can adjust them for Istio, instead of adjusting Istio for istio-csr.

Reply:
You make a good point, i think i was testing with an old version when i first set it up, i don't think even that `pilotSecretName` will change in the future so we could simply get rid of them both!

Reply:
You make a good point, i think i was testing with an old version when i first set it up, i don't think even that `pilotSecretName` will change in the future so we could simply get rid of them both!

On the previous point for documentation, what's the best place to write up all this information and a quick installation guide in the red hat openshift website or maistra docs?

Reply:
We will document this in the official OpenShift docs.

Reply:
Can i help in any way with that or are you good?

Reply:
It would be great if you could review our PRs related to this integration. I will request your reviews, ok?